src/release_notes_workflow/component_release_notes.py

```diff
@@ -0,0 +1,72 @@
+# SPDX-License-Identifier: Apache-2.0
+#
+# The OpenSearch Contributors require contributions made to
+# this file be licensed under the Apache-2.0 license or a
+# compatible open source license.
+
+import os
+from typing import Any
+from manifests.input_manifest import InputComponentFromSource
+
+
+class ComponentReleaseNotes:
+
+    def __init__(self, component: InputComponentFromSource, build_version: str, root: str) -> None:
+        self.component = component
+        self.build_version = build_version
+        self.root = root
+        self.filename
+
+    @property
+    def filename(self) -> str:
+        return self._filename
+
+    @filename.setter
+    def filename(self, filename): 
+        self._filename = filename
+
+    @property
+    def path(self) -> str:
+        return os.path.join(self.root, "release-notes")
+
+    def path_exists(self) -> bool:
+        return os.path.exists(self.path)
+
+    def exists(self) -> bool:
+        print(self.filename)
+        return self.path_exists() and any(fname.endswith(self.filename) for fname in os.listdir(self.path))
+
+
+    def check(self) -> str:
+        if self.exists():
+            return "YES"
+        else:
+            return "NO"
+
+
+class ComponentReleaseNotesOpenSearch(ComponentReleaseNotes):
+
+    @property
+    def filename(self) -> str:
+        return f'.release-notes-{self.build_version}.md'
+
+
+class ComponentReleaseNotesOpenSearchPlugin(ComponentReleaseNotes):
+
+    @property
+    def filename(self) -> str:
+        return f'.release-notes-{self.build_version}.0.md'
+
+
+class ComponentsReleaseNotes(ComponentReleaseNotes): 
+
+    @classmethod
+    def from_component(self, component: InputComponentFromSource, build_version: str, root: str) -> ComponentReleaseNotes:
+        if component.name == 'OpenSearch' or component.name == 'OpenSearch-Dashboards':
+            release_notes_os =  ComponentReleaseNotesOpenSearch(component, build_version, root)
+            ComponentReleaseNotes.filename = release_notes_os.filename
```

Comment:
This is wrong. If you need to “calculate” the filename differently in the different component classes create a constructor in each child class and do it in there.

Reply:
I have modified the logic to use property overloading.